javascript/javascript2/week3/script.js

```diff
@@ -0,0 +1,223 @@
+//Warmp
+//1
+
+const delayInMilliseconds = 2500;
+
+function delyaedText() {
+  setTimeout(function () {
+    const text = document.createElement("p");
+    const textNode = document.createTextNode(
+      "This text is called after 2.5 seconds"
+    );
+    document.getElementById("text-div").appendChild(textNode);
+  }, delayInMilliseconds);
+}
+
+delyaedText();
+
+//2
+
+function loggingOutString(delay, stringTolog) {
+  return function () {
+    console.log(delay, stringTolog);
+  };
+}
+
+const setForFiveSec = loggingOutString(5, "This string logged after 5 seconds");
+
+const setForThreeSec = loggingOutString(
+  3,
+  "This string logged after 3 seconds"
+);
+setForFiveSec();
+setForThreeSec();
+
+//3
+const btnShowText = document.getElementById("btn");
+
+btnShowText.addEventListener("click", function () {
+  setTimeout(function () {
+    console.log("called out after 5 seconds");
+  }, 5000);
+});
+
+//4
+function planetLogFuction(f) {
+  f();
+}
+function earthLogger() {
+  console.log("Earth");
+}
+function saturnLogger() {
+  console.log("Saturn");
+}
+planetLogFuction(earthLogger);
+planetLogFuction(saturnLogger);
+
+//5
+
+let pLocation = document.getElementById("location");
+
+function getLocation() {
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(getPosition);
+  } else {
+    pLocation.innerHTML = "Geolocation API is not Supported";
+  }
+}
+
+function getPosition(position) {
+  pLocation.innerHTML = `Latitude: ${position.coords.latitude} <br>longitude: ${position.coords.longitude}`;
+}
+
+//6
+//7
+function runAfterDelay(delay, callback) {
+  setTimeout(() => callback(), delay * 1000);
+  return delay;
+}
+
+runAfterDelay(1, function () {
+  console.log("should be logged after 1 seconds");
+});
+
+//8
+
+window.addEventListener("dblclick", (event) => {
+  setTimeout(() => event, 500);
+  console.log("Double-click detected");
+});
+
+//9
+function logFunny() {
+  return "this is a funny joke";
+}
+function logBad() {
+  return "this is a bad joke";
+}
+
+function jokeCreator(shouldTellFunnyJoke, logFunnyJoke, logBadJoke) {
+  if (shouldTellFunnyJoke) {
+    const funnyJoke = logFunnyJoke();
+    console.log(funnyJoke);
+  } else {
+    const badJoke = logBadJoke();
+    console.log(badJoke);
+  }
+}
+
+jokeCreator(true, logFunny, logBad);
+
+//2.Function as a variable
+
+let functionArray = [
+  firstFunctoin,
+  secondFunction,
+  thirdFunction,
+  function () {
+    console.log("annonmyse function");
+  },
```

Comment:
You misspelled "anonymous"! I can't spell either, but believe me, googling the right spelling is going to save so much time later on when you have so many variables and don't know why the code doesn't work just because you have one spelling in one file and another on the other file! IDEs usually have a spell checker, so check for this! ( I know it's a string here and it doesn't matter, but it got me in so much trouble when I have to program my website!)